packages/core/src/retry.ts

```diff
@@ -0,0 +1,30 @@
+interface RetryOptions {
+  retries?: number
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  onFailedAttempt?: (error: any, attemptCount: number) => PromiseLike<void> | void
+}
+
+export async function retry<T>(
+  input: (attemptCount: number) => PromiseLike<T> | T,
+  options?: RetryOptions
+): Promise<T> {
+  const retries = options?.retries ?? 2
```

Comment:
Move to a constant so it's clear?

```suggestion
  const retries = options?.retries ?? DEFAULT_RETRY_ATTEMPTS
```